lib/RGBot.js

```diff
@@ -463,87 +935,130 @@ const RGBot = class {
             matching: (block) => {
-                let blockFound = false;
-                if (blockType) {
-                    blockFound = (this.entityNamesMatch(blockType, block, { partialMatch }));
-                }
-                else if (block.type !== this.mcData.blocksByName.air.id) {
-                    blockFound = true; // if nothing specified... try anything but air
-                }
-                return blockFound;
-            },
-            useExtraInfo: (block) => {
-                if (onlyFindTopBlocks) {
-                    const blockAbove = this.#bot.blockAt(block.position.offset(0, 1, 0));
-                    return !blockAbove || blockAbove.type === 0 // only find if clear or 'air' above
+                const blockName = block.name
+                if (!blockNames || blockNames.length === 0 || (blockNames.includes(blockName)) || (partialMatch && blockNames.find(bn => bn.includes(blockName)))) {
+                    let blockDataForName = blockDataMap[blockName]
+                    if (!blockDataForName) {
+                        let canUseBlock = true;
+                        if (onlyFindTopBlocks) {
+                            const blockAbove = this.#bot.blockAt(block.position.offset(0, 1, 0));
+                            canUseBlock = !blockAbove || blockAbove.type === 0
+                        }
+                        if (canUseBlock) {
+                            const bestTool = this.bestHarvestTool(block)
+                            if (bestTool.digTime < Infinity) {
+                                blockDataForName = {
+                                    "value": blockValueFunction(blockName),
+                                    "tool": bestTool
+                                }
+                                blockDataMap[blockName] = blockDataForName
+                            }
+                        }
+                    }
+                    return blockDataForName && blockDataForName["value"] >= 0;
                 }
-                return true;
-            },
+            }
         });
 
-        let result = null;
-        if (nearbyBlocks.length > 0) {
-            if (!skipClosest) {
-                result = this.#bot.blockAt(nearbyBlocks[0]);
-            }
-            else if (nearbyBlocks.length > 1) {
-                result = this.#bot.blockAt(nearbyBlocks[1]);
-            }
+        for (let position of candidateBlockPositions) {
+            candidateBlocks.push(this.#bot.blockAt(position))
         }
 
-        if (result) {
-            this.#log(`Found Block of type ${blockType} within a maximum distance of ${maxDistance}`);
-        }
-        else {
-            this.#log(`I did not find any Blocks of type ${blockType} within a maximum distance of ${maxDistance}`);
-        }
-        return result;
+        this.#logDebug(`Detected ${Math.min(candidateBlocks.length, maxCount)} useful items to dig in range`)
```

Comment:
This is assuming that the user is using this function to find blocks to dig. I think we could probably just not have a log about this?